akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -316,6 +316,7 @@ private[persistence] trait Eventsourced extends Snapshotter with PersistenceStas
    * @param handler handler for each persisted `event`
    */
   def persist[A](event: A)(handler: A ⇒ Unit): Unit = {
+    if (recoveryRunning) throw new IllegalStateException("Cannot persist event during recovery")
```

Comment:
isn't the whole point to be able to persist events during recovery? I am confused here.